apps/src/applab/AppLabView.jsx

```diff
@@ -37,23 +49,63 @@ var AppLabView = React.createClass({
           onScreenCreate={this.props.onScreenCreate} />;
     }
 
+    var instructionsHeight = this.props.instructionsHeight;
+    if (this.props.instructionsCollapsed) {
+      instructionsHeight = HEADER_HEIGHT;
+    }
+
+    if (!this.props.instructionsInTopPane) {
+      instructionsHeight = 0;
+    }
```

Comment:
This bundle of logic for getting the 'real' `instructionsHeight` based on a few props would be a great helper method.